Packs/AWS-SecurityHub/Integrations/AWSSecurityHubEventCollector/AWSSecurityHubEventCollector.py

```diff
@@ -0,0 +1,264 @@
+import datetime as dt
+import urllib3
+from typing import Iterator
+
+import demistomock as demisto
+from AWSApiModule import *
+
+import boto3
+
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+VENDOR = 'AWS'
+PRODUCT = 'Security Hub'
+TIME_FIELD = 'CreatedAt'
+DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+DEFAULT_FIRST_FETCH = '3 days'
+DEFAULT_MAX_RESULTS = 1000
+
+
+def get_events(client: boto3.client, start_time: dt.datetime | None = None,
+               end_time: dt.datetime | None = None,
+               id_ignore_list: list[str] | None = None, limit: int = 0) -> Iterator[list[dict]]:
+    """
+    Fetch events from AWS Security Hub.
+
+    Args:
+        client (boto3.client): Boto3 client to use.
+        start_time (datetime | None, optional): Start time to fetch events from. Required if end_time is set.
+        end_time (datetime | None, optional): Time to fetch events until. Defaults to current time.
+        limit (int): Maximum number of events to fetch. Defaults to 0 (no limit).
+        id_ignore_list (list[str] | None, optional): List of finding IDs to not include in the results.
+            Defaults to None.
+
+    Yields:
+        tuple[list, CommandResults]: A tuple containing the events and the CommandResults object.
+    """
+    kwargs: dict = {'SortCriteria': [{'Field': TIME_FIELD, 'SortOrder': 'asc'}]}
+    filters: dict = {}
+
+    if end_time and not start_time:
+        raise ValueError('start_time must be set if end_time is used.')
+
+    if start_time:
+        filters[TIME_FIELD] = [{
+            'Start':
+                start_time.strftime(DATETIME_FORMAT),
+            'End':
+                end_time.strftime(DATETIME_FORMAT) if end_time else
+                dt.datetime.now().strftime('%Y-%m-%dT%H:%M:%SZ')
+        }]
+
+    if id_ignore_list:
+        ignore_filters = []
+        for event_id in id_ignore_list:
+            ignore_filters.append({'Value': event_id, 'Comparison': 'NOT_EQUALS'})
+
+        filters['Id'] = ignore_filters
+
+    if filters:
+        # We send kwargs because passing Filters=None to get_findings() tries to use a None value for filters,
+        # which raises an error.
+        kwargs['Filters'] = filters
+
+    count = 0
+
+    while True:
+        # The API only allows a maximum of 100 results per request. Using more raises an error.
+        if limit and limit - count < 100:
+            kwargs['MaxResults'] = limit - count
+
+        else:
+            kwargs['MaxResults'] = 100
```

Comment:
```suggestion
        kwargs['MaxResults'] = limit - count if limit and limit - count < 100 else 100
```

Reply:
Isn't it a lot less readable / understandable in a one-liner?

Reply:
it's your decision, but I think that in this case one-liner is better here